### ISSUE TITLE
Date formats are inconsistent

### ISSUE THREAD

**Issue by maxkueng** [restic/restic#1306]
The date formats from `restic snapshots` and `restic snapshots --json` and the fuse mount are inconsistent. The output from `restic snapshots` doesn't even show the UTC offset and doesn't have the "T" separator.

It'd be great if the all used the same format. And even better if they were displayed in UTC.

Since the formats of `restic snapshots` and the fuse mount are different there's no way to copy-paste the date and use it to browse the fuse mount.

## Output of `restic version`

restic 0.7.3
compiled with go1.9 on linux/amd64

## What backend/server/service did you use?

local file system

## Expected behavior

 - Date format in `restic snapshots --json`: 2017-10-01T17:36:25.988212963+02:00
 - Date format in `restic snapshots`: 2017-10-01T17:36:25+02:00
 - Date format in mounted repo: 2017-10-01T17:36:25+02:00

## Even better expected behavior

Would be even better if the date format was always in UTC.

 - Date format in `restic snapshots --json`: 2017-10-01T15:36:25.988212963Z
 - Date format in `restic snapshots`: 2017-10-01T15:36:25Z
 - Date format in mounted repo: 2017-10-01T15:36:25Z

## Actual behavior

 - Date format in `restic snapshots --json`: 2017-10-01T17:36:25.988212963+02:00
 - Date format in `restic snapshots`: 2017-10-01 17:36:25
 - Date format in mounted repo: 2017-10-01T17:36:25+02:00

## Steps to reproduce the behavior

Initialize repo:
```sh
RESTIC_PASSWORD=test restic -r /tmp/test_repo init
```

Back up a file:
```sh
RESTIC_PASSWORD=test restic -r /tmp/test_repo backup /tmp/testfile
```

List snapshots:
```sh
RESTIC_PASSWORD=test restic -r /tmp/test_repo snapshots
```

List snapshots with JSON output:
```sh
RESTIC_PASSWORD=test restic -r /tmp/test_repo snapshots --json
```

Create mount directory:
```sh
mkdir /tmp/mnt_test_repo
```

Mount repo:
```sh
RESTIC_PASSWORD=test restic -r /tmp/test_repo mount /tmp/mnt_test_repo
```

List snapshots in mounted repo:
```sh
ls -l /tmp/mnt_test_repo/snapshots
```

**Comment by fd0** [comment]
Thanks for the comprehensive report. I'd like to add the following considerations, timestamps in restic are used for different purposes:
 * In the `snapshots` command the timestamps are displayed so that they can be easily read by humans, which means using a proper formatting and displaying the times in the local timezone.
 * In the `snapshots` command with `--json`, the timestamps should be easily parsed by whatever consumes the JSON. Readability is not so important here.
 * In the fuse mount, the snapshot timestamp is used as a directory name, so it's best not to use any spaces.

I don't see how we can easily unify the timestamp formats for these different usage scenarios. I don't like just using the RFC format everywhere, that's very hard to read.

Do you have a proposal?

**Comment by ricardoseriani** [comment]
How about a flag to customize the timestamps ?

**Comment by maxkueng** [comment]
My use case is basically that I want to be able to copy the timestamp from `restic snapshots` and paste it in to my `cp` or `ls` or `cd` commands when browsing the mounted volume. With the current output from `restic snapshots` this requires manually replacing the space with a "T" and tabbing to add the "+02:00".

But when #1102 is implemented this will actually be less of an issue.

**Comment by paul-t-t** [comment]
I have a related suggestion here: it would be good to avoid (or have the option to avoid) colons in the timestamp format used for fuse mount directory names. 

I ran into a problem with this because I'm sharing the mounted repository as a SMB share which is then mounted by a MacOS client. The presence of colons in the directory names causes MacOS and/or Samba to mangle the names. 

You can configure Samba to apply a mapping to names to avoid this, but it's rather messy, and it would be good to avoid this if possible by using the compact form is ISO8601 -- e.g. 20171001T173625+0200. Simpler still would be to use UTC, so as to avoid timezone complexities: 20171001T153625. I agree it's not pretty, but it is maximally interoperable.

**Comment by fd0** [comment]
Huh, thanks for adding this point. I think it makes sense to make it configurable for the fuse mount at least.

**Comment by KalleDK** [comment]
I just tested to see how restic uses timezones - and as far as I can see the snapshot return the date in the timezone it was created.

These two was created with two different timezones - but in reallity only 30 min appart.
5330839a  2018-06-07 09:00:18  cloud01_bind                 /etc/bind
5d76958b  2018-06-07 11:29:28  cloud02_bind                 /etc/bind

My expectation would be that they either are marked with the timezone or are converted to my timezone (the one duing the snapshots command) or are alle showed in UTC.

I would ofcourse prefer the show times in my timezone - but I can understand force to UTC - Though I would not expect the user to "guess" the timezone the backup was created in

**Comment by KalleDK** [comment]
After a few minutes thought - I would say that the usage of local timezone in the show commands might not be the best idea. This is due to how forget works - as it might would cause inconsistency if the same forget is run in different timezones. So I would suggest to show only UTC. If local timezones should be used then there would be need for some concrete examples on how forget would react to repositories with backups from different timezones.

**Comment by fd0** [comment]
Good point about the `forget` command

**Comment by dimejo** [comment]
If restic always saves as UTC but converts to local timezone for displaying purpose only, would this still be problematic?

**Comment by KalleDK** [comment]
@dimejo this was my thought too - but the a more extensive documentation would be needed.. Sin the documentation says the last of the day is kept with keep-day... but if you are in another timezone you would see maybe the first of the day or one 2 hours before midnight etc.. This would might confuse people

**Comment by dimejo** [comment]
> @dimejo this was my thought too - but the a more extensive documentation would be needed.. Sin the documentation says the last of the day is kept with keep-day... but if you are in another timezone you would see maybe the first of the day or one 2 hours before midnight etc.. This would might confuse people

Ah, now I understand. That's indeed problematic.

**Comment by fd0** [comment]
You're right, this may be problematic and needs triage. Thanks for pointing it out!

**Comment by KalleDK** [comment]
My first intuition in solving this would change forget to not use UTC but use now() as the stating point. This is due to if you use UTC this would be, for some, a random time during the day - whereas now() would be when they start the command - therefore are you able to better control it yourself. This would also make snapshots able to show times in local timezone - as their timestamp is not "bound" to UTC day change.

**Comment by fd0** [comment]
FYI, I've recently looked into how `forget` works, see #2233
